python/cudf_polars/cudf_polars/callback.py

```diff
@@ -145,7 +146,31 @@ def _callback(
         set_device(device),
         set_memory_resource(memory_resource),
     ):
-        return ir.evaluate(cache={}).to_polars()
+        return ir.evaluate(cache={}, config=config).to_polars()
+
+
+def validate_config_options(config: dict) -> None:
+    """
+    Validate the configuration options for the GPU engine.
+
+    Parameters
+    ----------
+    config
+        Configuration options to validate.
+
+    Raises
+    ------
+    ValueError
+        If the configuration contains unsupported options.
+    """
+    if unsupported := (config.keys() - {"raise_on_fail", "parquet_options"}):
+        raise ValueError(
+            f"Engine configuration contains unsupported settings {unsupported}"
+        )
+    if parquet_options := config.get("parquet_options", {}):
+        assert {"chunked", "chunk_read_limit", "pass_read_limit"}.issuperset(
+            parquet_options
+        )
```

Comment:
The conditional isn't needed here since you default to the empty set anyway right?
```suggestion
    assert {"chunked", "chunk_read_limit", "pass_read_limit"}.issuperset(
        config.get("parquet_options", {})
    )
```